crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,894 @@
+    );
+    let dependency = make_esm_helpers_dependency(
+      options,
+      &asset_file_path,
+      asset_environment,
+      dependency_flags,
+      asset_id,
+    );
+    dependency_by_specifier.insert(dependency.specifier.as_str().into(), dependency);
+  }
+
+  if let Some(hoist_result) = result.hoist_result {
+    asset.flags.set(AssetFlags::HAS_SYMBOLS, true);
+    // Pre-allocate expected symbols
+    asset
+      .symbols
+      .reserve(hoist_result.exported_symbols.len() + hoist_result.re_exports.len() + 1);
+
+    // Collect all exported variable names into `asset.symbols`
+    for symbol in &hoist_result.exported_symbols {
+      let symbol = transformer_exported_symbol_into_symbol(&asset_file_path, &symbol);
+      asset.symbols.push(symbol);
+    }
+
+    // Collect all imported symbols into each of the corresponding dependencies' symbols array
+    for symbol in hoist_result.imported_symbols {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        let symbol = transformer_imported_symbol_to_symbol(&asset_file_path, &symbol);
+        dependency.symbols.push(symbol);
+      }
+    }
+
+    for symbol in hoist_result.re_exports {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        if is_re_export_all_symbol(&symbol) {
+          let loc = Some(convert_loc(asset_file_path.clone(), &symbol.loc));
+          dependency.symbols.push(make_export_all_symbol(loc));
+          // TODO: Why isn't this added to the asset.symbols array?
+        } else {
+          let existing = dependency
+            .symbols
+            .as_slice()
+            .iter()
+            .find(|candidate| candidate.exported == &*symbol.imported);
+          let existing_flags = existing
+            .map(|e| e.flags & SymbolFlags::IS_WEAK)
+            .unwrap_or(SymbolFlags::IS_WEAK);
+
+          // TODO: What is this? There is no such mangling in the transformer
+          let re_export_name = existing
+            .map(|sym| sym.local.clone())
+            .unwrap_or_else(|| format!("${:016x}$re_export${}", asset_id, symbol.local).into());
+          let symbol = Symbol {
+            exported: symbol.imported.as_ref().into(),
+            local: re_export_name.clone(),
+            loc: Some(convert_loc(asset_file_path.clone(), &symbol.loc)),
+            flags: existing_flags,
+          };
+
+          dependency.symbols.push(symbol.clone());
+          asset.symbols.push(symbol);
+        }
+      }
+    }
+
+    for specifier in hoist_result.wrapped_requires {
+      if let Some(dep) = dependency_by_specifier.get_mut(&swc_core::atoms::JsWord::new(specifier)) {
+        dep.flags.set(DependencyFlags::SHOULD_WRAP, true);
+      }
+    }
+
+    // for (name, specifier) in hoist_result.dynamic_imports {
+    //   if let Some(dep) = dependency_by_specifier.get_mut(&specifier) {
+    //     dep.promise_symbol = Some((&*name).into());
```

Comment:
What is `promise_symbol`? I suppose this is the mangled name inserted to replace `import(...)`?